### PR TITLE
feat(strings): simplify repeat

### PR DESCRIPTION
**Pull request by ecyrbe**
this simplifies repeat algorithm with a more redable one still using `O(log2(N))` algorithm.

### CODE REVIEW

**Comment by sno2** [comment]
Ah, nice idea on changing the repeat pattern itself instead of using accumulators. That looks so much better. You can create the PR for length as well if you would like. Otherwise, I can do it later tonight.

**Comment by ecyrbe** [comment]
@sno2 i still don't understand how the length algo works, so yes, go for it. or if you provide me a detailed explaination i can help improve it.

File: src/internals/numbers/impl/digits/division.ts [hunk 1]
```diff
@@ -59,20 +59,27 @@ export type DivModDigits<
...
 
 export type ModDigits<N extends Digit[], M extends Digit[]> = TruncateWith<
   N,
   M
 > extends [infer A extends Digit[], infer D extends Digit[]]
-  ? DivModDigits<A, D, M>["Remainder"]
+  ? _DivModDigits<A, D, M>["Remainder"]
+  : never;
+
+export type DivModDigits<N extends Digit[], M extends Digit[]> = TruncateWith<
```
Comment (sno2) [inline src/internals/numbers/impl/digits/division.ts:80]: Hello, not including the `M` type param here seems to have broken `src/internals/numbers/impl/digits/power.ts` on line 14.

Comment (ecyrbe) [reply src/internals/numbers/impl/digits/division.ts:80]: thanks, fixing it now

**Pull request merged**